pkg/api/rpms.go

```diff
@@ -59,6 +68,11 @@ type SearchRpmResponse struct {
 	Summary     string `json:"summary"`      // Summary of the package found
 }
 
+type DetectRpmsResponse struct {
+	Found   pq.StringArray `json:"found" gorm:"type:text"`   // List of rpm names found
+	Missing pq.StringArray `json:"missing" gorm:"type:text"` // List of rpm names not found in given repositories
+}
```

Comment:
I would change these types to []string, and create a new model to use in the dao layer for collecting these from the database. This would to be separate API concerns from database concerns.